### PR TITLE
Update cl_ext.h for cl_intel_queue_no_sync_operations

### PR DESCRIPTION
**Pull request by aharon-abramson**
Adding macros for cl_intel_queue_no_sync_operations

### CODE REVIEW

**Comment by aharon-abramson** [comment]
@bashbaug, can you please review?

File: CL/cl_ext.h
Comment (bashbaug) [inline CL/cl_ext.h:2447]: How much trouble would it be to choose a different bit number than "4" for this enum?

While there's no requirement to do this, for other bitfields (e.g. cl_mem_flags, cl_command_queue_properties) we've assigned bits for vendor extensions to higher bits (e.g. 31 on down), leaving the lower bits free for standard functionality.  I'd recommend continuing this convention if at all possible.

File: CL/cl_ext.h
Comment (bashbaug) [inline CL/cl_ext.h:2444]: A few cosmetic suggestions:

* Can you expand this "header comment" to follow a similar format as the other extensions?  Right now it's easy to misinterpret these defines as a part of the preceding extension and not as a new extension.
* Can you add a comment describing which bitfield type `CL_QUEUE_NO_SYNC_OPERATIONS` is for?  I think it's an addition to `cl_command_queue_properties`?

We really should merge the header generation script (#161) so most of this will happen automatically...

**Review by bashbaug — CHANGES_REQUESTED** [review]
Can you please also make a PR to update the XML grammar?

https://github.com/KhronosGroup/OpenCL-Docs/blob/main/xml/cl.xml

**Comment by bashbaug** [comment]
Note: it would be best to have a non-Intel review approval before merging.

File: CL/cl_ext.h
Comment (aharon-abramson) [reply CL/cl_ext.h:2447]: Sure, no problem.

File: CL/cl_ext.h
Comment (EwanC) [inline CL/cl_ext.h:2451]: why did you go for `27` out of interest? Wondering if there are other extensions that haven't reserved the xml value. `29` is the highest currently available bit position in https://github.com/KhronosGroup/OpenCL-Docs/blob/main/xml/cl.xml#L793

Comment (aharon-abramson) [reply CL/cl_ext.h:2451]: Yes, I got confused by CL_QUEUE_CAPABILITY_KERNEL_INTEL. I'll change it to 29.

File: CL/cl_ext.h
Comment (oddhack) [reply CL/cl_ext.h]: There's a weak process note about reserving enums in cl.xml, but nothing about bits AFAIK. Making that process stronger (Vulkan has a very detailed set of steps to be taken before consuming shared resources like bits) doesn't guarantee no collisions or assigning values from the wrong end, but it definitely helps a lot.

**Review by EwanC — APPROVED** [review]

**Pull request merged**
